src/Platforms/AbstractPlatform.php

```diff
@@ -1162,6 +1167,10 @@ public function getCreateSchemaSQL(string $schemaName): string
             throw NotSupported::new(__METHOD__);
         }
 
+        if (preg_match('/^\d/', $schemaName) > 0) {
```

Comment:
Even though it's not consistently documented, all `get[...]SQL()` methods of this class accept SQL as `string` parameters and just concatenate them with other fragments of SQL. Treating the schema name as a literal is against this implied contract. It's the caller of the method who should quote the names that need to be quoted.

Depending on how you manage your schema via DBAL, you may try explicitly quoting names in the schema definition similar to this: https://github.com/doctrine/dbal/blob/95d093a306e3cd059a5ddf2e461c24c6fddf5a4f/tests/Functional/Schema/OracleSchemaManagerTest.php#L251-L255

This won't be necessary in DBAL 5.0 (see https://github.com/doctrine/dbal/pull/6589).

Reply:
Within the current API, a proper solution would be to assume all object names obtained by introspecting the database as quoted, but schema names are returned as strings, so there's not way to indicate that besides wrapping all names in quotes. Doing that will fix this bug but will be a breaking change, so we cannot do that either.

Reply:
hi @morozov 

it's not really clear to me what to do to fix this? are our hands tied and this whole stuff (and handling postgresql schema in dbal) cannot be fixed before v5?

Reply:
As for fixing this issue, yes. But my understanding is that you want to fix a test failure that prevents the https://github.com/doctrine/dbal/pull/6576 from succeeding. I believe this issue needs to be addressed in the test suite, not in the code. For instance, you may improve isolation of the tests so that the lack of proper handling of schema names that need to be quoted doesn't affect the tests that don't use such names.